Helloworld/network.js

```diff
@@ -0,0 +1,169 @@
+    return this._formatMessage({
+      id,
+      default: id,
+      description: id,
+    })
+  }
+
+  getInfo() {
+    return {
+      id: 'HelloworldNetworkExt',
+      name: this.formatMessage('NetworkExt.ExtName'),
+      color1: '#e000ff',
+      blocks: [
+        {
+          opcode: 'httpget',
+          blockType: 'reporter',
+          text: this.formatMessage('NetworkExt.httpget'),
+          arguments: {
+            url: {
+              type: 'string',
+              defaultValue: 'https://extensions.turbowarp.org/hello.txt',
+            },
+            query: {
+              type: 'string',
+              defaultValue: undefined,
+            },
+          },
+        },
+        {
+          opcode: 'httppost',
+          blockType: 'reporter',
+          text: this.formatMessage('NetworkExt.httppost'),
+          arguments: {
+            url: {
+              type: 'string',
+              defaultValue: 'https://extensions.turbowarp.org/hello.txt',
+            },
+            body: {
+              type: 'string',
+              defaultValue: 'body',
+            },
+          },
+        },
+        {
+          opcode: 'encoding',
+          blockType: 'reporter',
+          text: this.formatMessage('NetworkExt.encoding'),
+          arguments: {
+            json: {
+              type: 'string',
+              defaultValue: '{"nihao":"666","wohao":"777"}',
+            },
+          },
+        },
+        {
+          opcode: 'decoding',
+          blockType: 'reporter',
+          text: this.formatMessage('NetworkExt.decoding'),
+          arguments: {
+            uri: {
+              type: 'string',
+              defaultValue: 'nihao=666&wohao=777',
+            },
+          },
+        }
+      ],
+    }
+  }
+  encoding(args)
+  {
+    let obj=JSON.parse(args.json);
+    let encodedString = Object.keys(obj).map(key => key + '=' + obj[key]).join('&');
+    return encodedString;
+  }
+  decoding(args)
+  {
+    let encodedString=args.uri;
+    let decodedObj = {};
+    encodedString.split('&').forEach(function(keyValue)
+    {
+        var pair = keyValue.split('=');
+        decodedObj[pair[0]] = pair[1];
+    });
+    return JSON.stringify(decodedObj);
+  }
+  httpget(args)
+  {
+    // console.log(args);
+    let xhr= new XMLHttpRequest();
+    let url=args.url;
+    if(args.query)
+    {
+      url+='?';
+      url+=args.query;
+    }
+    xhr.open('GET',url,false);
+    xhr.send(null);
+    return xhr.responseText;
+  }
+  httppost(args)
+  {
+    let xhr=new XMLHttpRequest();
+    xhr.open('POST',args.url,false);
+    xhr.send(args.body);
+    return xhr.responseText;
+  }
+}
+
+window.tempExt = {
+    Extension: Network,
+    info: {
+      name: 'NetworkExt.ExtName',
+      description: 'NetworkExt.description',
+      extensionId: 'HelloworldNetworkExt',
+      // iconURL: icon,
+      // insetIconURL: cover,
+      featured: true,
+      disabled: false,
+      collaborator: 'Helloworld@CCW',
```

Comment:
Recommendation:
HELLOWORD @ CCW
(add spaces where appropriate to improve readability)